libs/s25main/mapGenerator/Algorithms.h

```diff
@@ -147,6 +147,33 @@ namespace rttr { namespace mapGenerator {
      */
     void UpdateDistances(NodeMapBase<unsigned>& distances, std::queue<MapPoint>& queue);
 
+    /**
+     * Computes a map of distance values describing the distance of each grid position to the closest flagged point.
+     *
+     * @param flaggedPoints contains all flagged points
+     * @param size size of  the map
+     *
+     * @return distance of each grid position to closest flagged point.
+     */
+    template<class T_Container>
+    NodeMapBase<unsigned> DistancesTo(const T_Container& flaggedPoints, const MapExtent& size)
+    {
+        const unsigned maximumDistance = size.x * size.y;
+        std::queue<MapPoint> queue;
+        NodeMapBase<unsigned> distances;
+        distances.Resize(size, maximumDistance);
+
+        for(const MapPoint& pt : flaggedPoints)
+        {
+            distances[pt] = 0;
+            queue.push(pt);
+        }
```

Comment:
If you want you can check if this is even faster. I'd expect a bit of a speed penalty by pushing the elements one by one to the queue which means reallocation.

```suggestion
        NodeMapBase<unsigned> distances;
        distances.Resize(size, maximumDistance);
        std::queue<MapPoint> queue(flaggedPoints.begin(), flaggedPoints.end());

        for(const MapPoint& pt : flaggedPoints)
            distances[pt] = 0;
```

Reply:
what about this?

Reply:
I don't expect any significant improvements by this after  “Pass container with flagged points to distance calc function” brought only 0.1s and the current version mirrors the other impl of this function (and might be replaced in the future by that) so I see no benefit from this change